tests/source/test_interpolate.f90

```diff
@@ -0,0 +1,103 @@
+program test_interpolate
+    use math_constants
+    use util
+
+    implicit none
+
+    real(8), parameter :: TOL = 1.0d-6
+
+    real(8), parameter :: X_MIN = 0.0d0, X_MAX = TWOPI
```

Comment:
Test with something other than 0 so that the `x_min` mentioned elsewhere is also covered.